src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3494,13 +3626,11 @@ void checkLoopCondition()
                         AppendNewBlock(whenBothNotNull);
 
                         controlVariableReferenceforCondition = getLoopControlVariableReference(forceImplicit: true); // Yes we are going to evaluate it again
-                        limitReference = getCaptureReference(limitValueId, operation.LimitValue);
+                        limitReference = GetCaptureReference(limitValueId, operation.LimitValue);
 
-                        controlVariableReferenceforCondition = TryUnwrapNullableValue(controlVariableReferenceforCondition, compilation) ??
-                                                                    MakeInvalidOperation(((INamedTypeSymbol)operation.LimitValue.Type).TypeArguments[0], controlVariableReferenceforCondition);
-                        limitReference = TryUnwrapNullableValue(limitReference, compilation) ??
-                                                    MakeInvalidOperation(((INamedTypeSymbol)operation.LimitValue.Type).TypeArguments[0], limitReference);
-                        // PROTOTYPE(dataflow): The scenario with missing GetValueOrDefault is not covered by unit-tests.
```

Comment:
Prototype comment isn't addressed?

Reply:
> Prototype comment isn't addressed?

No

Reply:
> Prototype comment isn't addressed?

To clarify, it is not addressed, but the comment is moved into the helper that is supposed to deal with that. 

---
In reply to: [189756446](https://github.com/dotnet/roslyn/pull/26994#discussion_r189756446) [](ancestors = 189756446)